backend/main.py

```
@@ -107,38 +111,127 @@ async def startup_event():
+            # Keep connection alive and handle any incoming messages
+            data = await websocket.receive_text()
+            import json
+            message = json.loads(data)
+            
+            if message.get("type") == "ping":
+                await websocket.send_text(json.dumps({"type": "pong", "timestamp": time.time()}))
+            
+    except WebSocketDisconnect:
+        progress_manager.disconnect(session_id)
+        print(f"WebSocket {session_id} disconnected")
+    except Exception as e:
```

Ruff (0.13.3)

Do not catch blind exception: `Exception`

(BLE001)